core/controllers/collection_editor.py

```diff
@@ -221,10 +239,12 @@ def put(self, collection_id):
             else:
                 raise self.InvalidInputException(
                     'Cannot unpublish a collection.')
-
+        rights = rights_manager.get_collection_rights(
```

Comment:
Nope, call it rights_dict, to distinguish it from the domain object.

Actually I don't think you should modify it here. The backend rights domain object + dict is already a well-defined thing. The frontend should calculate the relevant properties when obtaining the corresponding frontend domain object.